usr/src/cmd/zdb/zdb.c

```diff
@@ -1558,8 +1558,32 @@ dump_deadlist(dsl_deadlist_t *dl)
 static avl_tree_t idx_tree;
 static avl_tree_t domain_tree;
 static boolean_t fuid_table_loaded;
-static boolean_t sa_loaded;
-sa_attr_type_t *sa_attr_table;
+sa_attr_type_t *sa_attr_table = NULL;
+
+static int
+check_sa_attr_table(objset_t *os)
+{
+	uint64_t sa_attrs = 0;
+	uint64_t version;
+	int error;
+
+	if (sa_attr_table != NULL)
```

Comment:
Too bad there isn't any way to assert that the existing sa_attr_table is for this objset.  Maybe we should add a global `objset_t *sa_os` and set it when we load the attr_table, just so that we can assert that we didn't forget to unload an old attr table?